cpp/ql/src/Likely Bugs/Memory Management/UsingExpiredStackAddress.ql

```diff
@@ -117,25 +137,27 @@ TGlobalAddress globalAddress(Instruction instr) {
     result = TLoad(globalAddress(load.getSourceAddress()))
   )
   or
-  exists(ConvertInstruction convert, Type fromType, Type toType | instr = convert |
-    uncheckedConversionTypes(convert, fromType, toType) and
-    result = TConversion("unchecked", globalAddress(convert.getUnary()), fromType, toType)
+  exists(Type fromType, Type toType, Instruction unary |
```

Comment:
Nit: I wonder is we could lift all the logic into the new module. So essentially the three blocks would read:

```codeql
Conversion<ConvertInstruction>::Using<uncheckedConversionTypes/3>::existInstruction("unchecked") or
Conversion<CheckedConvertOrNullInstruction>::Using<checkedConversionTypes/3>::existInstruction("checked") or
Conversion<InheritanceConversionInstruction>::Using<inheritanceConversionTypes/3>::existInstruction("inheritance")
```

Or something simillar, now we are already moving parts of the logic into a module parameterized with both the instruction type and the predicate to be used.

Reply:
We probably could. Note that we really only introduced this module to force a specific join order. So if we move more things into the module then we'll probably reintroduce the bad join order.